### PR TITLE
Fix inconsistencies between CO2 simulation and carbon simulation with CO2 only

### PR DESCRIPTION
**Pull request by msulprizio**
### Name and Institution (Required)

Name: Melissa Sulprizio
Institution: Harvard

### Describe the update

This pull request will address differences that still remain in CO2 between the traditional CO2 simulation and the carbon simulation with CO2 only. Fixes so far include:

1. Add missing ship logical to HEMCO_Config.rc.carbon, otherwise ship emissions for CO2 are not included
2. Update HEMCO_Config.rc.CO2 to:
    a. Include DiagnFile to properly save out emissions diagnostics
    b. Use same Aviation_SurfCorr_SclFac.1x1.nc file as carbon simulation

### Expected changes

This is a zero difference update with respect to the full chemistry benchmark. It will impact CO2 in the carbon simulations.

### Reference(s)

If this is a science update, please provide a literature citation.

### Related Github Issue

- https://github.com/geoschem/geos-chem/issues/2463

See also:
- https://github.com/geoschem/GCHP/issues/440
- https://github.com/geoschem/GCHP/issues/437
- https://github.com/geoschem/geos-chem/issues/2387

### CODE REVIEW

**Comment by msulprizio** [comment]
Additional updates and fixes now include:

- shunting the call to KPP in `carbon_gases_mod.F90` for CO2-only carbon simulations
- renaming subroutine `Emiss_Carbon_Gases` to `CO2_Production` to reflect that only CO2 production from CO are read and applied in that routine
- adding logical switches to `HEMCO_Confic.rc.carbon` to avoid reading data that is not needed in single-species carbon simulations

We are now able to reproduce CO2 concentrations between the CO2 simulation and CO2-only carbon simulations. See plots in #2463.

File: GeosCore/carbon_gases_mod.F90 [hunk 1]
```diff
@@ -166,7 +166,13 @@ SUBROUTINE Emiss_Carbon_Gases( Input_Opt,  State_Chm, State_Diag,            &
     !========================================================================
     ! CO2 production from CO oxidation
     !========================================================================
-    IF ( Input_Opt%LCHEMCO2 .and. id_CO2_adv > 0 ) THEN
+
+    ! If CO2 is an advected species but CO is not, use CO2_COPROD field
+    ! from HEMCO. Otherwise compute CO2 production via KPP.
+    IF ( id_CO2_adv > 0 .and. id_CO_adv <= 0 ) THEN
+
+       WRITE( 6, 100 )
```
Comment (yantosca) [inline GeosCore/carbon_gases_mod.F90:174]: @msulprizio: I think we should wrap the write statement in a `IF ( Input_Opt%amIRoot ) THEN ...` block.

Comment (yantosca) [reply GeosCore/carbon_gases_mod.F90:174]: Also not sure if that has to be written out every timestep or just on the 1st timestep.  I will defer to @lizziel.

**Review by yantosca — CHANGES_REQUESTED** [review]
Thanks @msulprizio.  I had a couple of minor comments but will defer to @lizziel.

File: CHANGELOG.md [hunk 2]
```diff
@@ -4,6 +4,15 @@ This file documents all notable changes to the GEOS-Chem repository starting in
...
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/), and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
+## [Unreleased] - TBD
+### Changed
+- Renamed `Emiss_Carbon_Gases` to `CO2_Production` in `carbon_gases_mod.F90`
+
+### Fixed
+- Added a fix to skip the call to KPP when only CO2 is defined in the carbon simulation
+- Added fix to turn on ship emissions for CO2 in the carbon simulation
+- Update `HEMCO_Config.rc` for carbon simulation to read data based on carbon species used
```
Comment (lizziel) [inline CHANGELOG.md:14]: Updated

File: GeosCore/carbon_gases_mod.F90 (hunk 1 above)
Comment (lizziel) [reply GeosCore/carbon_gases_mod.F90:174]: I agree to wrap in amIRoot and also to only print in first timestep

**Review by lizziel — CHANGES_REQUESTED** [review]

File: GeosCore/carbon_gases_mod.F90 (hunk 1 above)
Comment (msulprizio) [reply GeosCore/carbon_gases_mod.F90]: This has now been addressed in commit https://github.com/geoschem/geos-chem/pull/2510/commits/68604792ea1f4340f37d1d33bf1782c16e48e795.

File: CHANGELOG.md (hunk 2 above)
Comment (msulprizio) [reply CHANGELOG.md]: Now fixed in https://github.com/geoschem/geos-chem/pull/2510/commits/68604792ea1f4340f37d1d33bf1782c16e48e795

**Comment by msulprizio** [comment]
All GCClassic and GCHP integrations pass

**Comment by Aosrain** [comment]
@msulprizio When I try to recompile GCClassic with the updated codes, it shows the following error information.

=================================
make -j
[  1%] Built target KPP_FirstPass
[  1%] Built target HETP_core
[  4%] Built target HeadersHco
[  6%] Built target CloudJ_Core
[  6%] Built target JulDayHco
[ 16%] Built target Headers
[ 21%] Built target NcdfUtilHco
[ 23%] Built target JulDay
[ 24%] Built target GeosUtilHco
[ 29%] Built target NcdfUtil
[ 44%] Built target HCO
[ 47%] Built target GeosUtil
[ 49%] Built target ObsPack
[ 52%] Built target History
[ 67%] Built target HCOX
[ 69%] Built target KPP
[ 70%] Built target HCOI_Shared
[ 72%] Building Fortran object src/GEOS-Chem/GeosCore/CMakeFiles/GeosCore.dir/carbon_gases_mod.F90.o
[ 72%] Building Fortran object src/GEOS-Chem/GeosCore/CMakeFiles/GeosCore.dir/input_mod.F90.o
/nobackup/lwu5/GCClassic/GC_run/gc_4x5_merra2_carbon/CodeDir/src/GEOS-Chem/GeosCore/input_mod.F90(1601): error #6460: This is not a field name that is defined in the encompassing structure.   [AER_OPTICS_DIR]
    Input_Opt%AER_OPTICS_DIR = TRIM( v_str )
--------------^
/nobackup/lwu5/GCClassic/GC_run/gc_4x5_merra2_carbon/CodeDir/src/GEOS-Chem/GeosCore/input_mod.F90(1601): error #6303: The assignment operation or the binary expression operation is invalid for the data types of the two operands.   [TRIM]
    Input_Opt%AER_OPTICS_DIR = TRIM( v_str )
-------------------------------^
/nobackup/lwu5/GCClassic/GC_run/gc_4x5_merra2_carbon/CodeDir/src/GEOS-Chem/GeosCore/input_mod.F90(2837): error #6460: This is not a field name that is defined in the encompassing structure.   [CLOUD_FLAG]
    Input_Opt%Cloud_Flag = v_int
--------------^
/nobackup/lwu5/GCClassic/GC_run/gc_4x5_merra2_carbon/CodeDir/src/GEOS-Chem/GeosCore/input_mod.F90(2851): error #6460: This is not a field name that is defined in the encompassing structure.   [OD_INCREASE_FACTOR]
    Input_Opt%OD_Increase_Factor = Cast_and_RoundOff( v_str, places=4 )
--------------^
/nobackup/lwu5/GCClassic/GC_run/gc_4x5_merra2_carbon/CodeDir/src/GEOS-Chem/GeosCore/input_mod.F90(2865): error #6460: This is not a field name that is defined in the encompassing structure.   [MIN_CLOUD_OD]
    Input_Opt%Min_Cloud_OD = Cast_and_RoundOff( v_str, places=4 )
--------------^
/nobackup/lwu5/GCClassic/GC_run/gc_4x5_merra2_carbon/CodeDir/src/GEOS-Chem/GeosCore/input_mod.F90(2881): error #6460: This is not a field name that is defined in the encompassing structure.   [CLOUD_CORR]
    Input_Opt%Cloud_Corr = Cast_and_RoundOff( v_str, places=3 )
--------------^
/nobackup/lwu5/GCClassic/GC_run/gc_4x5_merra2_carbon/CodeDir/src/GEOS-Chem/GeosCore/input_mod.F90(2896): error #6460: This is not a field name that is defined in the encompassing structure.   [NUM_MAX_OVERLAP]
    Input_Opt%Num_Max_Overlap = v_int
--------------^
/nobackup/lwu5/GCClassic/GC_run/gc_4x5_merra2_carbon/CodeDir/src/GEOS-Chem/GeosCore/input_mod.F90(2913): error #6460: This is not a field name that is defined in the encompassing structure.   [SPHERE_CORRECTION]
    Input_Opt%Sphere_Correction = v_int
--------------^
/nobackup/lwu5/GCClassic/GC_run/gc_4x5_merra2_carbon/CodeDir/src/GEOS-Chem/GeosCore/input_mod.F90(2930): error #6460: This is not a field name that is defined in the encompassing structure.   [NUM_WV_BINS]
    Input_Opt%Num_WV_Bins = v_int
--------------^
/nobackup/lwu5/GCClassic/GC_run/gc_4x5_merra2_carbon/CodeDir/src/GEOS-Chem/GeosCore/input_mod.F90(2943): error #6460: This is not a field name that is defined in the encompassing structure.   [USE_H2O_UV_ABS]
    Input_Opt%USE_H2O_UV_Abs = v_bool
--------------^
/nobackup/lwu5/GCClassic/GC_run/gc_4x5_merra2_carbon/CodeDir/src/GEOS-Chem/GeosCore/input_mod.F90(2943): error #6303: The assignment operation or the binary expression operation is invalid for the data types of the two operands.   [V_BOOL]
    Input_Opt%USE_H2O_UV_Abs = v_bool
-------------------------------^
compilation aborted for /nobackup/lwu5/GCClassic/GC_run/gc_4x5_merra2_carbon/CodeDir/src/GEOS-Chem/GeosCore/input_mod.F90 (code 1)
make[2]: *** [src/GEOS-Chem/GeosCore/CMakeFiles/GeosCore.dir/build.make:452: src/GEOS-Chem/GeosCore/CMakeFiles/GeosCore.dir/input_mod.F90.o] Error 1
make[2]: *** Waiting for unfinished jobs....
make[1]: *** [CMakeFiles/Makefile2:1392: src/GEOS-Chem/GeosCore/CMakeFiles/GeosCore.dir/all] Error 2
make: *** [Makefile:136: all] Error 2

**Comment by msulprizio** [comment]
> All GCClassic and GCHP integrations pass

Correction to my previous statement: All GCClassic integration tests are passing. The GCHP carbon CO2 simulation failed, even though the integration test reported it as passing (separate issue reported in #2527). I am investigating the failure (in ExtData) and will push a fix to this PR before requesting reviews again.

**Comment by msulprizio** [comment]
GCClassic and GCHP integration tests are now indeed passing. I pushed additional fixes for CO2 to ExtData.rc to get the GCHP carbon_CO2 simulation to complete.

GCClassic differences for CO2 now look reasonable:

- [gcc_co2_sim_vs_gcc_carbon_co2.pdf](https://github.com/user-attachments/files/17497736/gcc_co2_sim_vs_gcc_carbon_co2.pdf)
- [gcc_co2_sim_vs_gcc_carbon.pdf](https://github.com/user-attachments/files/17497747/gcc_co2_sim_vs_gcc_carbon.pdf)

GCHP differences in CO2 or the carbon (all species) simulation vs carbon with CO2 only looks reasonable as well:

- [gchp_carbon_vs_gchp_carbon-CO2.pdf](https://github.com/user-attachments/files/17497784/gchp_carbon_vs_gchp_carbon-CO2.pdf)

There are still large differences between CO2 in GCClassic and GCHP carbon simulations (all species vs CO2-only):

- [gcc_carbon_vs_gchp_carbon.pdf](https://github.com/user-attachments/files/17497792/gcc_carbon_vs_gchp_carbon.pdf)
- [gcc_carbon-CO2_vs_gchp_carbon-CO2.pdf](https://github.com/user-attachments/files/17497795/gcc_carbon-CO2_vs_gchp_carbon-CO2.pdf)

Tagging @lizziel for any insight on the GCClassic vs GCHP differences.

**Comment by Aosrain** [comment]
I turned off all the settings in HEMCO_Config.rc and geoschem_config.yml. It still shows these hot spots.

**Review by lizziel — APPROVED** [review]

**Comment by msulprizio** [comment]
Thanks @lizziel @Aosrain. I will merge in these fixes for now and we can add additional fixes for CO2 in GCHP separately.

**Pull request merged**